bin/exec.ml

```diff
@@ -209,4 +265,40 @@ let term : unit Term.t =
       restore_cwd_and_execve (Common.root common) prog args env)
 ;;
 
+let term : unit Term.t =
+  let+ builder = Common.Builder.term
+  and+ context = Common.context_arg ~doc:{|Run the command in this build context.|}
+  and+ prog = Arg.(required & pos 0 (some Cmd_arg.conv) None (Arg.info [] ~docv:"PROG"))
+  and+ no_rebuild =
+    Arg.(value & flag & info [ "no-build" ] ~doc:"don't rebuild target before executing")
+  and+ args = Arg.(value & pos_right 0 Cmd_arg.conv [] (Arg.info [] ~docv:"ARGS")) in
+  (* TODO we should make sure to finalize the current backend before exiting dune.
+     For watch mode, we should finalize the backend and then restart it in between
+     runs. *)
```

Comment:
I know that the comment is from before but I don't understand what it means. What is a "backend" in this context? Why restart? The comment raises more questions than in answers.

Reply:
Dune's commit history would suggest that I wrote this line (over 2 years ago) but I was curious because it doesn't sound like the way I write. I tracked down the original commit that was squashed into my commit: https://github.com/ocaml/dune/commit/90cbf648efc8b8b5185aad1f54b019dd25ed09bc (in this PR: https://github.com/ocaml/dune/pull/6507). The "backend" @rgrinberg is referring to here is the `Console.Backend.t` which abstracts the way dune prints messages.

Reply:
The change described in that comment doesn't sound complicated so maybe I'll just make the suggested change (in another PR) while I remember how `dune exec` works.